stackslib/src/net/api/getstxtransfercost.rs

```diff
@@ -92,9 +119,58 @@ impl RPCRequestHandler for RPCGetStxTransferCostRequestHandler {
         _contents: HttpRequestContents,
         node: &mut StacksNodeState,
     ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
-        // todo -- need to actually estimate the cost / length for token transfers
-        //   right now, it just uses the minimum.
-        let fee = MINIMUM_TX_FEE_RATE_PER_BYTE;
+        let estimated_len = self
+            .estimated_len
+            .take()
+            .ok_or(NetError::SendError("`estimated_len` not set".into()))?;
+
+        let fee_resp = node.with_node_state(|_network, sortdb, _chainstate, _mempool, rpc_args| {
+            let tip = self.get_canonical_burn_chain_tip(&preamble, sortdb)?;
+            let stacks_epoch = self.get_stacks_epoch(&preamble, sortdb, tip.block_height)?;
+
+            if let Some((_, fee_estimator, metric)) = rpc_args.get_estimators_ref() {
+                // STX transfer transactions have zero runtime cost
+                let estimated_cost = ExecutionCost::zero();
+                let estimations =
+                    RPCPostFeeRateRequestHandler::estimate_tx_fee_from_cost_and_length(
+                        &preamble,
+                        fee_estimator,
+                        metric,
+                        estimated_cost,
+                        estimated_len,
+                        stacks_epoch,
+                    )?
+                    .estimations;
+                if estimations.len() != 3 {
+                    // logic bug, but treat as runtime error
+                    return Err(StacksHttpResponse::new_error(
+                        &preamble,
+                        &HttpServerError::new(
+                            "Logic error in fee estimation: did not get three estimates".into(),
+                        ),
+                    ));
+                }
+
+                // safety -- checked estimations.len() == 3 above
+                let median_estimation = &estimations[1];
+
+                // NOTE: this returns the fee _rate_
+                Ok(median_estimation.fee / estimated_len)
```

Comment:
This fee / byte rate does seem to match what the previous implementation did, but I'm not sure how meaningful that value is or if users know how to use that rate properly. I'd definitely like to hear input from users of this endpoint (wallets) on this.

Reply:
The [documentation](https://github.com/stacks-network/stacks-core/blob/master/docs/rpc-endpoints.md#get-v2feestransfer) says its a fee per byte.

That said, I'm regretting the `length=` query parameter, since it won't even be used.  I'll just take that out.